File: src/main/java/com/violeth/blockjs/blockjs/binds/RegionBind.java
package com.violeth.blockjs.blockjs.binds;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

/** Optional WorldGuard region support
 * TODO: It was just a prototype */
public class RegionBind {
    public String regionName;

    public List<File> scripts;

    public RegionBind(String regionName, File script) {
        this.regionName = regionName;

        this.scripts = new ArrayList<>();

        this.scripts.add(script);
    }
}

File: plugin/src/main/kotlin/io/github/blockjs/plugin/binds/RegionBind.kt
package io.github.blockjs.plugin.binds

import java.io.File

/** Optional WorldGuard region support
 * TODO: It was just a prototype  */
class RegionBind(
    var regionName: String,
    script: File,
    scripts: MutableList<File> = mutableListOf()
) {
    init {
        scripts.add(script)
    }
}
